pkg/filestore/filestore.go

```diff
@@ -172,13 +174,13 @@ func (upload *fileUpload) GetReader(ctx context.Context) (io.ReadCloser, error)
 }
 
 func (upload *fileUpload) Terminate(ctx context.Context) error {
-	if err := os.Remove(upload.infoPath); err != nil {
-		return err
-	}
-	if err := os.Remove(upload.binPath); err != nil {
+	// Account for server crashes or force kills
+	// Say if killed just after removing the bin file
+	err := os.Remove(upload.binPath)
+	if !os.IsNotExist(err) {
 		return err
 	}
-	return nil
+	return os.Remove(upload.infoPath)
```

Comment:
We should also ignore errors when the file was already deleted here and return `nil` in these cases.

Reply:
I think it may not be needed as that would be an illegal call - because both the files would have been already deleted.

Reply:
The underlying files might be removed by a cron job for cleaning up and in those cases `Terminate` should not return an error because the files are already gone. This is likely an edge case, but since we are already touching this logic it makes sense to do it properly.